package.json

```
@@ -4,17 +4,26 @@
   "description": "The secure record transfer application has been designed for the purpose of transferring files and standardized metadata over the internet to an institution.",
   "scripts": {
     "build": "webpack --config webpack.config.js && node scripts/compress_images.js",
-    "watch": "webpack --watch --config webpack.config.js"
+    "watch": "webpack --watch --config webpack.config.js",
+    "lint": "eslint '**/*.{js,cjs,mjs}'"
```

Running `npm run lint` on Windows fails with this error:

```
> bagitobjecttransfer@2024.10.22 lint
> eslint '**/*.{js,cjs,mjs}'


Oops! Something went wrong! :(

ESLint: 9.17.0

No files matching the pattern "'**/*.{js,cjs,mjs}'" were found.
Please check for typing mistakes in the pattern.
```

I also realized that if you're targeting everything matching that pattern, you're going to lint these files which is not the desired behaviour:

- `node_modules/`
- `env/` (JS files in Python virtual environment)
- `dist/` (Built static resources)
- `bagitobjecttransfer/static/` (Collected static resources)

This suggestion works in tandem with another change I'm requesting in this review:

```suggestion
    "lint": "eslint"
```

---

> Running `npm run lint` on Windows fails with this error:
> 
> ```
> > bagitobjecttransfer@2024.10.22 lint
> > eslint '**/*.{js,cjs,mjs}'
> 
> 
> Oops! Something went wrong! :(
> 
> ESLint: 9.17.0
> 
> No files matching the pattern "'**/*.{js,cjs,mjs}'" were found.
> Please check for typing mistakes in the pattern.
> ```

I think this could be because single quotes are not recognized as a way to wrap strings in Windows. It I change them to escaped double quotes as shown below, it should work:

```
"lint": "eslint \"**/*.{js,cjs,mjs}\""
```
I agree with your suggested change because it looks like `eslint` already lints files with the above 3 extensions by default, so specifying them explicitly is redundant:
https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-without-extension

---

Including these directories in `ignores` makes sense to me:
> * `node_modules/`
> * `env/` (JS files in Python virtual environment)

However, I think these two would only be relevant to ignore if we want to be able run `eslint` in-container. Do you think this would be useful to have?
> * `dist/` (Built static resources)
> * `bagitobjecttransfer/static/` (Collected static resources)

If so, we'll have to include `eslint.config.mjs` as well when copying over files to the container in `Containerfile`, and list `eslint` dependencies under `dependencies` instead of `devDependencies` in `package.json`.

---

> However, I think these two would only be relevant to ignore if we want to be able run `eslint` in-container. Do you think this would be useful to have?
> 
> > * `dist/` (Built static resources)
> > * `bagitobjecttransfer/static/` (Collected static resources)
> 
> If so, we'll have to include `eslint.config.mjs` as well when copying over files to the container in `Containerfile`, and list `eslint` dependencies under `dependencies` instead of `devDependencies` in `package.json`.

I ran into issues with those two directories *outside* of a container. Consider these two situations that would necessitate those two directories being ignored:

1. Someone runs `npm run build` locally. I've personally done this to test out the webpack config, so it's conceivable that someone else might as well.
  - This would require `dist/` being in the ignore list
2. Someone runs `python manage.py collectstatic` locally. This is also something I've done when testing out the webpack config.
  - This would require `bagitobjecttransfer/static/` being in the ignore list

I think those directories are a safe bet to include in the case someone might possibly run ESLint after executing either of those two commands locally.

---

Ah I see, I didn't think of that - thanks for clarifying!